ext/CudaExt.jl

```diff
@@ -0,0 +1,60 @@
+module CudaExt
+
+import CUDA
+import ClimaComms: SingletonCommsContext, CUDADevice
+import ClimaTimeSteppers: compute_fj!
+
+@inline function compute_fj!(f, j, U, f!, j!, ::SingletonCommsContext{CUDADevice})
+    # TODO: we should benchmark these two options to
+    #       see if one is preferrable over the other
+    if Base.Threads.nthreads() > 1
+        compute_fj_spawn!(f, j, U, f!, j!)
+    else
+        compute_fj_streams!(f, j, U, f!, j!)
+    end
+end
+
+@inline function compute_fj_streams!(f, j, U, f!, j!)
+    event = CUDA.CuEvent(CUDA.EVENT_DISABLE_TIMING)
+    CUDA.record(event, CUDA.stream()) # record event on main stream
+
+    stream1 = CUDA.CuStream() # make a stream
```

Comment:
It could be that CUDA doesn't like making these streams at every RHS evaluation. Maybe we can try caching this in the timestepper?